### PR TITLE
Add `pre-commit` version update check

### LINKED ISSUE #382: Task: Bandit fails with API client errors
### What is the task about?

`make precommit` fails on bandit (v1.7.5) .

If those are new vulnerabilities, we could address them.

```

Test results:
>> Issue: [B113:request_without_timeout] Requests call without timeout
   Severity: Medium   Confidence: Low
   CWE: CWE-400 (https://cwe.mitre.org/data/definitions/400.html)
   More Info: https://bandit.readthedocs.io/en/0.0.0/plugins/b113_request_without_timeout.html
   Location: repository_service_tuf/helpers/api_client.py:50:23
49	        if method == Methods.get:
50	            response = requests.get(
51	                f"{server}/{url}", json=payload, data=data, headers=headers
52	            )
53	

--------------------------------------------------
>> Issue: [B113:request_without_timeout] Requests call without timeout
   Severity: Medium   Confidence: Low
   CWE: CWE-400 (https://cwe.mitre.org/data/definitions/400.html)
   More Info: https://bandit.readthedocs.io/en/0.0.0/plugins/b113_request_without_timeout.html
   Location: repository_service_tuf/helpers/api_client.py:55:23
54	        elif method == Methods.post:
55	            response = requests.post(
56	                f"{server}/{url}", json=payload, data=data, headers=headers
57	            )
58	

--------------------------------------------------
>> Issue: [B113:request_without_timeout] Requests call without timeout
   Severity: Medium   Confidence: Low
   CWE: CWE-400 (https://cwe.mitre.org/data/definitions/400.html)
   More Info: https://bandit.readthedocs.io/en/0.0.0/plugins/b113_request_without_timeout.html
   Location: repository_service_tuf/helpers/api_client.py:279:19
278	        console.print(f"Fetching file {file_uri}")
279	        response = requests.get(file_uri)
280	        if response.status_code != 200:

--------------------------------------------------
```

### Parent feature

_No response_

### References

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by KAUTH**
# Description

Add `pre-commit` version update check. For more info see https://github.com/repository-service-tuf/repository-service-tuf-cli/issues/382#issuecomment-1718443936.

Closes #382 

# Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)

# Additional requirements
- [ ] Tests have been added for the bug fix or new feature
- [x] Docs have been added for the bug fix or new feature

# Code of Conduct
By submitting this PR, you agree to follow our [Code of Conduct](CODE_OF_CONDUCT.rst).

- [x] I agree to follow this project's Code of Conduct

### CODE REVIEW

File: repository_service_tuf/helpers/api_client.py [hunk 1]
```diff
@@ -272,7 +280,7 @@ def send_payload(
 def get_md_file(file_uri: str) -> Metadata:
     if file_uri.startswith("http"):
         console.print(f"Fetching file {file_uri}")
-        response = requests.get(file_uri)
+        response = requests.get(file_uri, timeout=300)
```
Comment (MVrachev) [inline repository_service_tuf/helpers/api_client.py:283]: Why do you decide to add this?

**Review by MVrachev — APPROVED** [review]
Seems like a logical change.
@kairoaraujo do you agree?

If we agree on this we need to do that in the other repositories as well.

File: repository_service_tuf/helpers/api_client.py (hunk 1 above)
Comment (KAUTH) [reply repository_service_tuf/helpers/api_client.py:283]: I added timeouts to mitigate bandit's `B113: request_without_timeout`.

**Comment by KAUTH** [comment]
@kairoaraujo let's discuss here your comments on the way to go.

**Comment by MVrachev** [comment]
@kairoaraujo do you have other comments or can I merge this?

**Comment by kairoaraujo** [comment]
> @kairoaraujo let's discuss here your comments on the alternative way to go.

My point was that if a new contributor gets an outdated pre-commit they will still need to fix it.
I think we should do it separately in a PR.

For example, a new contributor fixes the documentation and they will get a problem from the check which is not related to the change.

**Comment by KAUTH** [comment]
> My point was that if a new contributor gets an outdated pre-commit they will still need to fix it. I think we should do it separately in a PR.
> 
> For example, a new contributor fixes the documentation and they will get a problem from the check which is not related to the change.

We have the same issue with updating the requirements. In `tox` we have a `requirements` env which checks if `make requirements` is up to date. If a package is outdated, once again it could require fixing before proceeding. In this sense, both cases are the same. `pre-commit autoupdate` is just updating a different set of requirements.
I think if we remove this one then we should be removing also the requirements check in tox.

What we could _additionally_ have is a new GH workflow to periodically update pre-commit versions (because we are not updating dependencies that are part of the source code I think it's also fine to do with CI (as in this PR), but then again it's not a big deal to add it, just an extra automated PR).

**Comment by kairoaraujo** [comment]
> @kairoaraujo We have the same issue with updating the requirements. In `tox` we have a `requirements` env which checks if `make requirements` is up-to-date. If a package is outdated, once again it could require fixing before proceeding (e.g., breaking change). In this sense, both cases are the same. `pre-commit autoupdate` is just updating a different set of requirements. I think if we remove this one then we should be removing also the requirements check in tox.

Yes, I know we do it on requirements, but I think we should fix it also.

IMO think we should not force the contributor to fix the requirements or pre-commit if it is not linked to their PR.
An example is where the contributor updated the docs and they need to deal with requirements, pre-commit.
My suggestion is to remove it from the check, maybe we can make the check write a comment in the PR as a nit, if the contributor has experience he can fix it, but not enforce it.


> What we could _additionally_ have is a new GH workflow to periodically update pre-commit versions (because we are not updating dependencies that are part of the source code I think it's also fine to do with CI (as in this PR), but then again it's not a big deal to add it, just an extra automated PR).

👍

**Review by kairoaraujo — APPROVED** [review]

**Review by kairoaraujo — APPROVED** [review]

**Pull request merged**